mqtt/src/main/scala/Mqtt.scala

```diff
@@ -31,6 +31,13 @@ private[mqtt] trait MqttConnectorLogic { this: GraphStageLogic =>
 
   def connectionSettings: MqttConnectionSettings
   def handleConnection(client: IMqttAsyncClient): Unit
+
+  /**
+   * Callback, that is called from the MQTT client thread before invoking
+   * message handler callback in the GraphStage context.
+   */
+  def beforeHandleMessage(): Unit
```

Comment:
When I think of it this entire trait is just for incoming rather than generic, right?